google-auth Made as optional dependency

shivanand007 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind feature
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
Makes Google auth as optional dependency, gives users a option to install it separately.

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

roycaihw commented on setup.py:
```diff
@@ -28,7 +28,8 @@
 EXTRAS = {
-    'adal': ['adal>=1.0.2']
+    'adal': ['adal>=1.0.2'],
+    'google-auth': ['google-auth>=1.0.0']
```
Shall we specify `google-auth>=1.0.1`, to be consistent with the requirements.txt removal?

shivanand007 replied:
> Shall we specify `google-auth>=1.0.1`, to be consistent with the requirements.txt removal?

Yeah, That's fine we can make it consistent to version 1.0.1

shivanand007 replied:
That change has been done

roycaihw reviewed (commented):
Please add a release note with instructions of how to install google-auth going forward.

We should also make sure the tests pass. I wonder if we need to update the test pipeline to install google-auth as well. 

/retest

shivanand007 commented:
> Please add a release note with instructions of how to install google-auth going forward.
> I have added the required release note, kindly have a review for the same.

> We should also make sure the tests pass. I wonder if we need to update the test pipeline to install google-auth as well.
> Yes, As we have Removed google-auth from requirements.txt, This must be a dependency in code which makes test cases to fail, 
    I will check the code and see if we can make those code blocks optional as well, 
    Additionally, we have update the test pipeline to test this use case.
> /retest

shivanand007 commented:
@roycaihw 
I have made the necessary changes in code to make google-auth as optional, Requesting to Change Test Pipeline, So That we can pass all the E2E test Cases.

shivanand007 commented:
/retest

mindw commented:
/remove-lifecycle rotten

roycaihw commented:
Sorry for the late reply. 

/ok-to-test

roycaihw commented:
/retest
